### PR TITLE
Dgrid feedback

### PR DESCRIPTION
**Pull request by maier49**
**Type:** bug + feature

**Description:** 

Fixes an issue where the source type of a `QueryTransformResult` isn't able to be queried, even though it was queried to create the result, adds the `FetchResult` functionality to the `QueryTransformResult`, extends the `FetchResult` interface to include a separate `dataLength` property, and adds documentation on the `FetchResult` type.

Please review this checklist before submitting your PR:

* [ ] There is a related issue
* [x] All contributors have signed a CLA
* [x] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [x] The code passes the CI tests
* [x] Unit or Functional tests are included in the PR
* [x] The PR increases or maintains the overall unit test coverage percentage
* [x] The code is ready to be merged

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -377,6 +377,56 @@ filteredView.observe().subscribe((update) => {
...
 
+
+### Fetch Results
+
+Both the `Store` and `QueryTransformResult` interfaces return a type called a `FetchResult` from `fetch`.
+This is a `Promise` that resolves to the fetched data, but it also has two other properties: `totalLength` and `dataLength`.
+For both the `Store` and `QueryTransformResult`, `totalLength` is a `Promise` that resolves to the total number of items
+in the underlying `Storage`.
+
+For a `Store`, `dataLength` resolves to the same value, and is only provided for consistency between the interfaces.
```
Comment (dylans) [inline README.md]: I would suggest changing "same value" to "same value as `totalLength`"

File: README.md [hunk 2]
```diff
@@ -377,6 +377,56 @@ filteredView.observe().subscribe((update) => {
...
+const queryResultWithoutQuery = queryResult.fetch();
+// This filter will not change the value of dataLength or totalLength
+const queryResultsWithQuery = queryResult.fetch(createFilter<any>().lessThan('value', 2));
+Promise.all([
+    queryResultWithoutQuery.totalLength,
+    queryResultsWithQuery.totalLength,
+    queryResultWithoutQuery.dataLength,
+    queryResultsWithQuery.dataLength
+]).then((values) => {
+    // values[0] === values[1] === 3 The total length in both cases is still the total number of items
```
Comment (dylans) [inline README.md]: totalLength rather than total length?

File: README.md [hunk 3]
```diff
@@ -377,6 +377,56 @@ filteredView.observe().subscribe((update) => {
...
+    // values[2] === values[3] === 2 The data length in both cases is the number of items matching the
```
Comment (dylans) [inline README.md]: dataLength rather than data length?

File: src/store/createQueryTransformResult.ts [hunk 4]
```diff
@@ -679,33 +680,52 @@ export const createQueryTransformResult: QueryTransformResultFactory = compose<Q
...
+		let resolveDataLength: Function;
+		let rejectDataLength: Function;
+		const dataLength = new Promise((resolve, reject) => {
+			resolveDataLength = resolve;
+			rejectDataLength = reject;
+		});
+		const fetchResult = state.source.fetch(firstQuery);
+		const resultsPromise = fetchResult.then(
+			(newData: any[]) => {
+				// We should apply the query transform result's own queries  first so that the total size of the locally
```
Comment (dylans) [inline src/store/createQueryTransformResult.ts]: Extra space before the word first

File: src/store/createQueryTransformResult.ts [hunk 5]
```diff
@@ -679,33 +680,52 @@ export const createQueryTransformResult: QueryTransformResultFactory = compose<Q
...
+				// cached data can be determined
+				newData = queryAndTransformData(queriesAndTransformations, newData);
+				resolveDataLength(newData.length);
+
+				if (mapped) {
+					const ids = mapped.identify(newData);
+					const newIndex = buildIndex(ids);
+					// Update this way if this is not an initial fetch. If this is the initial fetch, then this
+					// data(or subsequent data) will already be provided to observers in the initial notification, so don't
```
Comment (dylans) [inline src/store/createQueryTransformResult.ts]: Add space before (or

**Review by dylans — CHANGES_REQUESTED** [review]
Very minor nits, code changes look clear to me.

**Comment by maier49** [comment]
@dylans Nits have been picked

**Review by dylans — APPROVED** [review]

**Pull request merged**